plutus-contract/src/Language/Plutus/Contract/App.hs

```diff
@@ -99,7 +98,7 @@ printTrace con wllt ctr = do
             let st = newState previous
                 newRequest = ContractRequest { oldState = st, event = evt }
             BSL.putStrLn (Aeson.encode newRequest)
-            either (error . show) pure (insertAndUpdateContract con newRequest)
+            pure (insertAndUpdateContract con newRequest)
 
-    initial <- either (error . show) pure (initialiseContract @s con)
+    initial <- pure (initialiseContract @s con)
```

Comment:
Also missing rethrowing? not sure, maybe this is just being handled by the ambient effects now.

Reply:
Probably just worth checking all of these again.

Reply:
will do.

Reply:
In this case (`printTrace`) the new behaviour is cleaner: Instead of erroring out we print the whole result incl. the error, logs & event history.